src/mqueryfront/src/indexFiles/IndexPage.js

```diff
@@ -0,0 +1,148 @@
+import { Component } from "react";
+import IndexMultiSelect from "./IndexMultiSelect";
+import { useParams } from "react-router-dom";
+import api from "../api";
+import ErrorBoundary from "../components/ErrorBoundary";
+import IndexClearQueueButton from "./IndexClearQueueButton";
+
+function getAvailableTaintsListFromDatasets(datasets) {
+    var taintList = Object.values(datasets)
+        .map((ds) => ds.taints)
+        .flat();
+    return [...new Set(taintList)];
+}
+
+class IndexPageInner extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            filenames: [],
+            availableNGrams: ["gram3", "text4", "wide8", "hash4"],
+            selectedNGrams: [],
+            availableTaints: [],
+            selectedTaints: [],
+            size: 0,
+            oldestFile: null,
+            newestFile: null,
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleTextareaInput = this.handleTextareaInput.bind(this);
+        this.handleNGramSelect = this.handleNGramSelect.bind(this);
+        this.handleTaintSelect = this.handleTaintSelect.bind(this);
+    }
+
+    ursa_id = this.props.params.ursa_id;
+
+    componentDidMount() {
+        api.get(`/queue/${this.props.params.ursa_id}`)
+            .then((response) => {
+                this.setState({
+                    size: response.data.size, // NOTE: dunno whether these params will be provided at all
```

Comment:
```suggestion
                    size: response.data.size, // NOTE: I don't know whether these params will be provided at all
```
(sorry for hyporcisy if informal comments like this are elsewhere in the codebase :) )
Alternatively "these params can be null"